### PR TITLE
Add the favorite feature to the listing view

### PR DESCRIPTION
**Pull request by Edouard-6**

### CODE REVIEW

**Review by NervzZ — CHANGES_REQUESTED** [review]
Need to add test and also figure out why it's not passing the automated testing. I find a bit surprising that no method needs to be passed for the favorite button as it may make things less generic and maybe some other already existing composables could have been used instead.

File: app/src/main/java/com/studhub/app/presentation/listing/details/DetailedListingScreen.kt [hunk 1]
```diff
@@ -1,10 +1,7 @@
 package com.studhub.app.presentation.listing.details
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/listing/details/DetailedListingScreen.kt:1]: I don't see where you implement the favorite button in the screen?

File: app/src/main/java/com/studhub/app/presentation/listing/details/DetailedListingViewModel.kt [hunk 2]
```diff
@@ -2,28 +2,33 @@ package com.studhub.app.presentation.listing.details
...
-import com.studhub.app.presentation.listing.add.FakeListingRepository
+import com.studhub.app.domain.usecase.user.AddFavoriteListing
+import com.studhub.app.domain.usecase.user.GetFavoriteListings
+import com.studhub.app.domain.usecase.user.RemoveFavoriteListing
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.MutableSharedFlow
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/listing/details/DetailedListingViewModel.kt:16]: Duplicate import statements

File: app/src/main/java/com/studhub/app/presentation/listing/details/DetailedListingViewModel.kt [hunk 3]
```diff
@@ -32,4 +37,40 @@ class DetailedListingViewModel @Inject constructor(
             }
         }
     }
+
+    fun getFavorites() {
+        when (currentListing) {
+            is ApiResponse.Success -> {
+                val listing = (currentListing as ApiResponse.Success<Listing>).data
+                viewModelScope.launch {
+                    getFavoriteListings().collect {
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/listing/details/DetailedListingViewModel.kt:46]: import kotlinx.coroutines.flow.collect missing

File: app/src/main/java/com/studhub/app/presentation/listing/details/components/FavoriteButton.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package com.studhub.app.presentation.listing.details.components
+
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/listing/details/components/FavoriteButton.kt:2]: Remove this useless import

File: app/src/main/java/com/studhub/app/presentation/listing/details/components/FavoriteButton.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Favorite
+import androidx.compose.material.icons.filled.FavoriteBorder
+import androidx.compose.material3.Icon
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/listing/details/components/FavoriteButton.kt:6]: Replace with androidx.compose.material3.icons.Icons

**Review by raynasr — CHANGES_REQUESTED** [review]
Import statements to fix. LGTM.

**Review by raynasr — APPROVED** [review]

**Pull request closed without merging**